src/storage/checkpoint_io.cpp

```diff
@@ -0,0 +1,96 @@
+#include "storage/checkpoint_io.h"
+#include <catalog/schema.h>
+#include <vector>
+
+namespace terrier::storage {
+
+void AsyncBlockWriter::Open(const char *log_file_path, int buffer_num) {
+  out_ = PosixIoWrappers::Open(log_file_path, O_WRONLY | O_CREAT, S_IRUSR | S_IWUSR);
```

Comment:
Do you guys needs read permissions? I don't think AsyncBlockWriter reads anything from the file so you would only need write permissions.

Reply:
Fixed

Reply:
I guess you miss understood this permission. The `S_IRUSR` permission is stored in the filesystem metadata. For the file descriptor itself, it is indeed write only (`O_WRONLY`). If the permission of the file itself is write-only, nobody could read the content of that file. So how can someone recover it later?